root/static/scripts/release-editor/validation.js

```diff
@@ -123,13 +125,36 @@ $(function () {
 });
 
 
+// Search releases with the same barcode
+function searchExistingBarcode(field, barcode) {
+    request({
```

Comment:
I preferred to split the logic in a new function since the barcode validation block is already complex